pi5/board.go

```diff
@@ -162,16 +152,109 @@ func (b *pinctrlpi5) Reconfigure(
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	// make sure every pin has a name. We already know every pin has a pin
+	for _, c := range newConf.Pins {
+		if c.Name == "" {
+			c.Name = c.Pin
+		}
+	}
+
+	if err := b.addUserDefinedNames(newConf); err != nil {
+		return err
+	}
+
 	if err := b.reconfigurePullUpPullDowns(newConf); err != nil {
 		return err
 	}
+	if err := b.reconfigureInterrupts(newConf); err != nil {
+		return err
+	}
+	b.pinConfigs = newConf.Pins
+
+	return nil
+}
+
+// reconfigureInterrupts reconfigures the digital interrupts based on the new configuration provided.
+// It reuses existing interrupts when possible and creates new ones if necessary.
+func (b *pinctrlpi5) reconfigureInterrupts(newConf *rpiutils.Config) error {
+	// look at previous interrupt config, and see if we removed any
+	for _, oldConfig := range b.pinConfigs {
+		if oldConfig.Type != rpiutils.PinInterrupt {
+			continue
+		}
+		sameInterrupt := false
+		for _, newConfig := range newConf.Pins {
+			if newConfig.Type != rpiutils.PinInterrupt {
+				continue
+			}
+			// check if we still have this interrupt
+			if oldConfig.Name == newConfig.Name && oldConfig.Pin == newConfig.Pin {
+				sameInterrupt = true
+				break
+			}
+		}
+		// if we still have the interrupt, don't modify it
+		if sameInterrupt {
+			continue
+		}
+		// we no longer want this interrupt, so we will remove it and add back the pin's gpio functionality
```

Comment:
how do we know we don't still have the interrupt at this point? isn't it possible the name has just changed?

Reply:
yeah if the name was changed I go ahead and teardown the old interrupt, because the user wanted that pin to be reconfigured